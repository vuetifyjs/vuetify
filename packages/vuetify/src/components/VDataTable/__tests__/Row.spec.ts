// @ts-nocheck
/* eslint-disable */

// import Row from '../Row'
import {
  mount,
  Wrapper,
  MountOptions,
} from '@vue/test-utils'
// import Vue from 'vue'

describe.skip('Table Row', () => {
  type Instance = InstanceType<typeof Row>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(Row, options)
    }
  })

  it('should render without slots', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
          item: {
            petrol: 0.68,
            diesel: 0.65,
          },
        },
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render non-string values', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { value: 'string' },
            { value: 'number' },
            { value: 'array' },
            { value: 'boolean' },
            { value: 'object' },
            { value: 'undefined' },
            { value: 'null' },
          ],
          item: {
            string: 'string',
            number: 12.34,
            array: [1, 2],
            boolean: false,
            object: { foo: 'bar' },
            null: null,
          },
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with cellClass', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol', cellClass: 'a' },
            { text: 'Diesel', value: 'diesel', cellClass: ['b', 'c'] },
          ],
          item: {
            petrol: 0.68,
            diesel: 0.65,
          },
        },
      },
    })

    const tds = wrapper.findAll('td')
    expect(tds.at(0).classes()).toContain('a')
    expect(tds.at(1).classes()).toContain('b')
    expect(tds.at(1).classes()).toContain('c')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it.skip('should render with regular slots', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
        },
      },
      slots: {
        'column.petrol': '<p class="test">$0.68</p>',
        'column.diesel': '<p class="test">$0.65</p>',
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.findAll('p.test')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it.skip('should render with scoped slots', () => {
    const vm = new Vue()
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
          item: {
            petrol: 0.68,
            diesel: 0.65,
          },
        },
      },
      scopedSlots: {
        'column.petrol': props => vm.$createElement('p', { staticClass: `test ${props.header.value}` }, [props.value]),
        'column.diesel': props => vm.$createElement('p', { staticClass: `test ${props.header.value}` }, [props.value]),
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.findAll('p.test')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
