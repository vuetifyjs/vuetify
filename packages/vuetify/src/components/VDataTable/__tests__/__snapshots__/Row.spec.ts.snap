// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row should render non-string values 1`] = `
<tr>
  <td class="text-start">
    string
  </td>
  <td class="text-start">
    12.34
  </td>
  <td class="text-start">
    1,2
  </td>
  <td class="text-start">
    false
  </td>
  <td class="text-start">
    [object Object]
  </td>
  <td class="text-start">
  </td>
  <td class="text-start">
  </td>
</tr>
`;

exports[`Table Row should render with cellClass 1`] = `
<tr>
  <td class="text-start a">
    0.68
  </td>
  <td class="text-start b c">
    0.65
  </td>
</tr>
`;

exports[`Table Row should render without slots 1`] = `
<tr>
  <td class="text-start">
    0.68
  </td>
  <td class="text-start">
    0.65
  </td>
</tr>
`;
